mpactpy/core.py

```diff
@@ -0,0 +1,277 @@
+    quarter_sym_opt : QuarterSymmetryOption
+        Quarter core centerline symmetry
+        (i.e. whether the centerline bisects an assembly through the
+        center, or passes between assemblies along the edge)
+
+    Attributes
+    ----------
+    symmetry_opt : SymmetryOption
+        Core symmetry ("360", "90")
+    quarter_sym_opt : QuarterSymmetryOption
+        Quarter core centerline symmetry
+        (i.e. whether the centerline bisects an assembly through the
+        center, or passes between assemblies along the edge)
+    nx : int
+        Number of modules along the x-dimension
+    ny : int
+        Number of modules along the y-dimension
+    nz : int
+        Number of modules along the z-dimension
+    height : float
+        The total height of the core (cm)
+    mod_dim : Assembly.ModDim
+        The x,y,z dimensions of the ray-tracing module
+    assembly_map : List[List[Assembly]]
+        2-D map of the core assemblies
+    assemblies : List[Assembly]
+        The unique assemblies of this core
+    lattices : List[Lattice]
+        The unique lattices of this core
+    modules : List[Module]
+        The unique modules of this core
+    pins : List[Pin]
+        The unique pins of this core
+    pinmeshes : List[PinMesh]
+        The unique pin meshes of this core
+    materials : List[Material]
+        The unqiue materials of this core
+    """
+
+    SymmetryOption = Literal["360", "90", ""]
+    QuarterSymmetryOption = Literal["EDGE", "CENT", ""]
+
+    @property
+    def symmetry_opt(self) -> SymmetryOption:
+        return self._symmetry_opt
+
+    @property
+    def quarter_sym_opt(self) -> QuarterSymmetryOption:
+        return self._quarter_sym_opt
+
+    @property
+    def nx(self) -> int:
+        return len(self.assembly_map)
+
+    @property
+    def ny(self) -> int:
+        return len(self.assembly_map[0])
+
+    @property
+    def nz(self) -> int:
+        return self.assemblies[0].nz
+
+    @property
+    def height(self) -> float:
+        return self.assemblies[0].height
+
+    @property
+    def mod_dim(self) -> Assembly.ModDim:
+        return self.assemblies[0].mod_dim
+
+    @property
+    def assembly_map(self) -> List[List[Assembly]]:
+        return self._assembly_map
+
+    @property
+    def assemblies(self) -> List[Assembly]:
+        return self._assemblies
+
+    @property
+    def lattices(self) -> List[Lattice]:
+        return self._lattices
+
+    @property
+    def modules(self) -> List[Module]:
+        return self._modules
+
+    @property
+    def pins(self) -> List[Pin]:
+        return self._pins
+
+    @property
+    def pinmeshes(self) -> List[PinMesh]:
+        return self._pinmeshes
+
+    @property
+    def materials(self) -> List[Material]:
+        return self._materials
+
+
+    def __init__(self,
+                 assembly_map: List[List[Assembly]],
+                 symmetry_opt: SymmetryOption = "",
+                 quarter_sym_opt: QuarterSymmetryOption = ""):
```

Comment:
Minor thing, but my preference would be to use the MPACT defaults to be explicit and remove `""` from the Literals.
```suggestion
                 symmetry_opt: SymmetryOption = "360",
                 quarter_sym_opt: QuarterSymmetryOption = "EDGE"):
```

Reply:
I'm going to leave this as-is for now.  I'd rather keep `""` as an acceptable symmetry option since it is in MPACT.